domain/model/state/modelstate_test.go

```diff
@@ -323,6 +323,51 @@ INSERT INTO space (uuid, name) VALUES
 	c.Check(getCons, jc.DeepEquals, cons)
 }
 
+// TestSetModelConstraintsNullBools is a regression test for constraints to
+// specifically assert that allocate public ip address can be null, false and
+// true according to what the user wants.
+//
+// DQlite has a bug where null bool columns are reported back in select
+// statements as false even thought the value in the database is NULL. To get
+// around this bug we have updated the constraint table to strict and changed
+// the type on "allocate_public_ip" to an integer.
+func (s *modelSuite) TestSetModelConstraintsNullBools(c *gc.C) {
```

Comment:
Great test! Once you rebase this will be the only change in the PR but this is very good.